sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -196,6 +205,25 @@ def transform_messages_to_sendable_if_needed(messages):
         except AttributeError:
             return messages
 
+def create_messages_from_dicts_if_needed(messages, message_type):
+    """
+    This method is used to convert dict representations
+    of messages and to a list of ServiceBusMessage objects or ServiceBusBatchMessage.
+    """
+    # type: (DictMessageType) -> Union[List[azure.servicebus.ServiceBusMessage], azure.servicebus.ServiceBusMessageBatch]
+    if isinstance(messages, list):
+        for index, message in enumerate(messages):
+            if isinstance(message, dict):
+                messages[index] = message_type(**message)
+
+    if isinstance(messages, dict):
+        temp_messages = message_type(**messages)
+        messages = [temp_messages]
+
+    if isinstance(messages, message_type):
+        messages = [messages]
+
+    return messages
```

Comment:
We only deal with type of `dict` and `ServiceBusMessage` here and wrap the value into list. We could be more consistent on the returning type here.
```suggestion
    return [message_type(**messages)] if isinstance(messages, dict) else [messages]
```